option.go

```diff
@@ -0,0 +1,244 @@
+			pxe := dhcpv4.Options{
+				// PXE Boot Server Discovery Control - bypass, just boot from filename.
+				6:  []byte{8}, // or []byte{8}
+				69: binaryTpFromContext(ctx),
+			}
+			d.UpdateOption(dhcpv4.OptGeneric(dhcpv4.OptionVendorSpecificInformation, pxe.ToBytes()))
+		}
+	}
+
+	return withNetboot
+}
+
+// bootfileAndNextServer returns the bootfile (string) and next server (net.IP).
+func (s *Server) bootfileAndNextServer(ctx context.Context, mac net.HardwareAddr, uClass UserClass, opt60, bin string, tftp netaddr.IPPort, ipxe, iscript *url.URL) (string, net.IP) {
+	var nextServer net.IP
+	var bootfile string
+	if s.OTELEnabled {
+		span := trace.SpanContextFromContext(ctx)
+		tID := span.TraceID().String()
+		sID := span.SpanID().String()
+		fID := span.TraceFlags().String()
+		bin = fmt.Sprintf("%s-00-%v-%v-%v", bin, tID, sID, fID)
+	}
+	// If a machine is in an ipxe boot loop, it is likely to be that we aren't matching on IPXE or Tinkerbell userclass (option 77).
+	switch { // order matters here.
+	case uClass == Tinkerbell, (s.UserClass != "" && uClass == s.UserClass): // this case gets us out of an ipxe boot loop.
+		bootfile = "/no-ipxe-script-defined"
+		if iscript != nil {
+			bootfile = iscript.String()
+		}
+	case clientType(opt60) == httpClient: // Check the client type from option 60.
+		bootfile = fmt.Sprintf("%s/%s/%s", ipxe, mac.String(), bin)
+		ns := net.ParseIP(ipxe.Host)
+		if ns == nil {
+			s.Log.Error(fmt.Errorf("unable to parse ipxe host"), "ipxe", ipxe.Host)
+			ns = net.ParseIP("0.0.0.0")
+		}
+		nextServer = ns
+	case uClass == IPXE: // if the "iPXE" user class is found it means we aren't in our custom version of ipxe, but because of the option 43 we're setting we need to give a full tftp url from which to boot.
+		bootfile = fmt.Sprintf("tftp://%v/%v/%v", tftp.String(), mac.String(), bin)
+		nextServer = tftp.UDPAddr().IP
+	default:
+		bootfile = filepath.Join(mac.String(), bin)
+		nextServer = tftp.UDPAddr().IP
+	}
+
+	return bootfile, nextServer
+}
+
+// arch returns the arch of the client pulled from DHCP option 93.
+func arch(d *dhcpv4.DHCPv4) iana.Arch {
+	// get option 93 ; arch
+	fwt := d.ClientArch()
+	if len(fwt) == 0 {
+		return iana.Arch(255) // unknown arch
+	}
+	var archKnown bool
+	var a iana.Arch
+	for _, elem := range fwt {
+		if !strings.Contains(elem.String(), "unknown") {
+			archKnown = true
+			// Basic architecture identification, based purely on
+			// the PXE architecture option.
+			// https://www.iana.org/assignments/dhcpv6-parameters/dhcpv6-parameters.xhtml#processor-architecture
+			a = elem
+			break
+		}
+	}
+	if !archKnown {
+		return iana.Arch(255) // unknown arch
+	}
+
+	return a
+}
+
+// binaryTpFromContext extracts the binary trace id, span id, and trace flags
+// from the running span in ctx and returns a 26 byte []byte with the traceparent
+// encoded and ready to pass in opt43.
```

Comment:
These comments say opt43 but It looks like it's being passed in opt69 in option.go line 150.
I am confused.